js/rank_states_utils.js

```diff
@@ -71,80 +87,84 @@ function rankEm() {
+    .attr('alignment-baseline','hanging')
     .classed('bar-name',true)
     .classed('open-bar-name', function(d){
       return d.open;
     })
-    .attr('dy',"1em")
+    .attr('dy',"0.33em")
     .text(function(d){
       return d.abrv;
     });
   
   barGroups.append('rect')
     .attr('height', function(d){
-      return d.wu;
+      return y(d.wu) - 20;
     })
-    .attr('width','20')
+    .attr('width', barWidth)
     .style("stroke-dasharray","4, 2")
     .classed('open-rank-bar', function(d){
       return d.open;
     })
     .classed('closed-rank-bar', function(d){
       return !d.open;
     })
-    .style('fill', function(d){
-      if (!d.open){
-        return categoryToColor("total");
-      }
-    })
-    .on('mouseover', function(d){
-      if (d.open){
-        d3.select(this)
-          .style("stroke-dasharray",null)
-          .classed('chosen-rank-bar',true)
-          .style("fill", categoryToColor("total"));
-      }
-      })
-    .on('mouseout', function(d){
-      if (d.open){
-        d3.select(this)
-          .style("stroke-dasharray","4, 2")
-          .classed('chosen-rank-bar',false);
-        } 
-      });
+    .style('fill', categoryToColor("total"));
   
     function dragging(d) {
+      var thisShape = d3.select(this);
+      thisShape.attr("transform", "translate(" + (d3.event.x) + "," + (d3.event.y) + ")");
+      overRankBar(thisShape.node().getBoundingClientRect());
+    }
+
+    function dragdone(d) {
+      var barchoice = d3.select('.chosen-rank-bar');
+      if (barchoice.empty()){
+        d3.select(this)
+          .transition().duration(600)
+            .attr('transform','translate(0,0)'); 
+      } else {
+        barchoice
+          .on('mouseover', null)
+          .on('mouseout', null)
+          .attr('class','closed-rank-bar');
+        d3.select(barchoice.node().parentNode).select('text')
+          .classed('open-bar-name',false);
+        d3.select(this).style('opacity', 0)
+          .attr('transform','scale(0)'); 
+      }
+      overRankBar( d3.select(this).node().getBoundingClientRect()); // one last time in cast it overlapped two rectangles
       d3.select(this).attr("transform", "translate(" + (d3.event.x) + "," + (d3.event.y) + ")");
   }
-
-  function dragdone(d) {
-    var barchoice = d3.select('.chosen-rank-bar');
-    if (barchoice.empty()){
-      d3.select(this)
-        .transition().duration(600)
-          .attr('transform','translate(0,0)'); 
+  function overRankBar(shapeBox){
+  
+  var openBars = d3.selectAll('.open-rank-bar').nodes();
+  for (var i = 0; i < openBars.length; i++) { 
+    var thisBar = openBars[i];
+    var barBox = thisBar.getBoundingClientRect();
+    //console.log('testing overlap on bar '+bbox.right+' vs '+x);
+    var isOverBar = !(barBox.right < shapeBox.left || 
+                barBox.left > shapeBox.right || 
+                barBox.bottom < shapeBox.top || 
+                barBox.top > shapeBox.bottom);
```

Comment:
is this all to take care of being just barely outside of the rectangle?

Reply:
no, this is to test whether the state's bounding box is overlapping with one of the rectangles that is "open". My last PR had this taking advantage of the bar's hover event, but I changed it to this so it would also work on mobile.